machine-learning/_posts/2018-10-18-UsefulThingsML.md

```diff
@@ -0,0 +1,193 @@
+
+* Strong false assumptions can be better than weak true ones, because a
+learner with the latter needs more data to avoid overfitting.
+* **Regularization** functions added to the evaluation function help ensuring
+that the method explains the data with the simplest possible model.
+* **Cross-validation** can help to mitigate overfitting.
+* A common misconception about overfitting is that it is caused by noise.
+* Control the fraction of falsely accepted non-null hypotheses, known as the
+_false discovery rate_.
+
+
+5. **INTUITION FAILS IN HIGH DIMENSIONS**
+
+The _curse of dimensionality_ refers to either the fact that methods that work
+fine in low dimensions become intractable when the input is high-dimensional,
+or to the fact that Generalizing correctly becomes exponentially harder as
+the dimensionality (number of features) of the examples grows.
+
+Our intuitions, which come from a threedimensional world, often do not apply
+in high-dimensional ones.
+
+
+6. **THEORETICAL GUARANTEES ARE NOT WHAT THEY SEEM**the
+
+_Machine learning papers are full of theoretical guarantees_.
+
+The main role of theoretical guarantees in machine learning is not as a
+criterion for practical decisions, but as a source of understanding and
+driving force for algorithm desig
+
+Just because a learner has a theoretical justification and works in practice
+practice doesn't mean the former is the reason for the latter.
+
+
+7. **FEATURE ENGINEERING IS THE KEY**
+
+At the end of the day, some machine learning projects succeed and some fail.
+What makes the difference? Easily the most important factor is the features
+used. **Often, the raw data is not in a form that is amenable to learning, but
+you can construct features from it that are**.
+
+Machine learning is not a one-shot process of building a data set and running
+a learner, but rather an iterative process of running the learner, analyzing
+the results, modifying the data and/or the learner, and repeating.
+
+**Features that look irrelevant in isolation may be relevant in combination**.
+
+
+8. **MORE DATA BEATS A CLEVERER ALGORITHM**
+
+Pragmatically the quickest path to success is often to just get more data. As
+a rule of thumb, a dumb algorithm with lots and lots of data beats a clever
+one with modest amounts of it.
+
+Enormous mountains of data are available, but there is not enough time to
+process it, so it goes unused. Even though in principle more data means that
+more complex classifiers can be learned, **in practice simpler classifiers end
+up being used, because complex ones take too long to learn**. **Part of the
+answer is to come up with fast ways to learn complex classifiers**.
+
+**Part of the reason using cleverer algorithms has a smaller payoff than you
+might expect is that, to a first approximation, they all do the same**.
+
+**Because of the curse of dimensionality**, no existing amount of data may be
+enough. For these reasons, **clever algorithms often pay off in the end**.
+

```

Comment:
Doesn't that contradict the whole section ?

Reply:
As Antoine said, the last sentence seem to contradict the section. The way I understand it, is that when you don't have lots of data, encoding priors into your classifier is key. Also it seems to say that you basically never have enough data. Which seems like an unsupported claim to me, and makes the whole thing confusing.

I think we can summarize it like this:
* If you can get more data, do it (instead of changing your classifier)
* If you have very few samples and you can't get much more (e.g. medical imaging), you should put more effort on giving priors to your classifier.

Reply:
I do share that `no existing amount of data may be enough.` seems to convey the idea that no solution gives an accurate-enough answer. A model may not have a 100% accuracy, but as long as it beats the formers, I think it is good enough so as to deserve some room in the literature.

So yes, re-formatted the section and incorporated your comments in 9bd2c4c.